src/sidebar/components/tag-editor.js

```diff
@@ -88,98 +106,97 @@ function TagEditor({ onEditTags, tags: tagsService, tagList }) {
+    if (item) {
+      addTag(item);
     }
   };
 
-  const handleKeyUp = e => {
-    // Safari on macOS and iOS have an issue where pressing "Enter" in an
-    // input when its value exactly matches a suggestion in the associated <datalist>
-    // does not generate a "keypress" event. Therefore we catch the subsequent
-    // "keyup" event instead.
+  /**
+   * Opens the Datalist on focus if there is a value in the input
+   */
+  const handleFocus = () => {
+    if (inputEl.current.value.trim().length) {
+      setSuggestionsListOpen(true);
+    }
+  };
 
-    if (e.key === 'Enter') {
-      // nb. `addTag` will do nothing if the "keypress" event was already handled.
-      addTag();
+  /**
+   *  Called when the user uses keyboard navigation to move
+   *  up or down the suggestions list creating a highlighted
+   *  item.
+   *
+   *  The first value in the list is an unselected value (-1).
+   *  A user can arrive at this value by pressing the up arrow back to
+   *  the beginning or the down arrow until the end.
+   *
+   * @param {number} direction - Pass 1 for the next item or -1 for the previous
+   */
+  const changeSelectedItem = direction => {
+    let nextActiveItem = activeItem + direction;
+    if (nextActiveItem < -1) {
+      nextActiveItem = suggestions.length - 1;
+    } else if (nextActiveItem >= suggestions.length) {
+      nextActiveItem = -1;
     }
+    setActiveItem(nextActiveItem);
   };
 
-  const suggestionsList = () => {
-    // If this is Chrome, use the filtered list, otherwise use the full list
-    // See https://github.com/hypothesis/client/issues/1606
-    const optionsList = isChromeUA ? suggestionsFiltered : suggestions;
-    return (
-      <datalist
-        id={datalistId}
-        className="tag-editor__suggestions"
-        aria-label="Annotation suggestions"
-      >
-        {showSuggestions &&
-          optionsList.map(suggestion => (
-            <option key={suggestion} value={suggestion} />
-          ))}
-      </datalist>
-    );
+  /**
+   * Keydown handler for keyboard navigation of the suggestions list
+   * and when the user presses "Enter" or ","" to add a new typed item not
+   * found in the suggestions list
+   */
+  const handleKeyDown = e => {
+    switch (e.key) {
+      case 'ArrowUp':
+        changeSelectedItem(-1);
+        e.preventDefault();
+        break;
+      case 'ArrowDown':
+        changeSelectedItem(1);
+        e.preventDefault();
+        break;
+      case 'Enter':
+      case ',':
+        if (activeItem === -1) {
+          // nothing selected, just add the typed text
+          addTag(inputEl.current.value);
+        } else {
+          addTag(suggestions[activeItem]);
+        }
+        e.preventDefault();
+        break;
+    }
   };
 
+  // Only add the aria-activedescendant prop when there is an item selected.
+  // -1 value means no item selected.
+  const activeDescendant =
+    activeItem >= 0 ? `${tagEditorId}-datalist-item-${activeItem}` : false;
```

Comment:
In testing I found that VoiceOver with Safari and Chrome did not read the suggestion when pressing the down/up arrow for the first time to open the list, but did read subsequent suggestions after that. I changed the false branch here so that the `aria-activedescendant` attribute is set when there is no active item, but with an empty/missing value, and that resolved the problem.

```suggestion
    activeItem >= 0 ? `${tagEditorId}-datalist-item-${activeItem}` : "";
```

Reply:
Interested, I totally do not experience the same behavior, but I'll make that change anyway, It would be good to know why we're seeing a difference :\